dockstore-webservice/src/main/java/io/dockstore/webservice/core/Collection.java

```diff
@@ -151,15 +155,15 @@ public void setDescription(String description) {
     }
 
     public void setEntries(Set<Entry> entries) {
-        this.entries = entries;
+        this.entries = entries.stream().map(EntryVersion::new).collect(Collectors.toSet());
     }
 
-    public void addEntry(Entry entry) {
-        this.entries.add(entry);
+    public void addEntry(Entry entry, Version version) {
+        this.entries.add(new EntryVersion(entry, version));
     }
 
-    public void removeEntry(Entry entry) {
-        this.entries.remove(entry);
+    public void removeEntry(Entry entry, Long versionId) {
+        this.entries.removeIf(entryVersion -> entryVersion.getEntry().getId() == (entry.getId()) && ((entryVersion.getVersion() != null ? entryVersion.getVersion().getId() == versionId : null == versionId)));
```

Comment:
This looks re-usable
Might be worth storing in EntryVersion as a comparator like https://github.com/dockstore/dockstore/blob/e8b4a6760787c9890817ac9a44b53d98bcb31f7d/dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ObsoleteUrlFactory.java#L119 or 

https://github.com/dockstore/dockstore/blob/develop/dockstore-webservice/src/main/java/io/dockstore/webservice/core/FileFormat.java#L48